### ISSUE TITLE
amp-story: Zooming obscures content

### ISSUE THREAD

**Issue by TetraLogicalHelpdesk** [ampproject/amphtml#32503]
# Zooming obscures content

## WCAG Level
Level AA

## Priority
High

## Pages/screens/components affected
* [Navigation](https://www.gstatic.com/amphtml/stamp/qa/navigation.html)
* [RTL](https://www.gstatic.com/amphtml/stamp/qa/rtl.html)
* [Video](https://www.gstatic.com/amphtml/stamp/qa/video.html)
* [Audio - amp-story background audio](https://www.gstatic.com/amphtml/stamp/qa/audio-story.html)
* [Audio - amp-story-page background audio](https://www.gstatic.com/amphtml/stamp/qa/audio-page.html)
* [Audio - amp-story using amp-audio](https://www.gstatic.com/amphtml/stamp/qa/amp-audio.html)
* [Animation](https://www.gstatic.com/amphtml/stamp/qa/animation.html)
* [Sharing](https://www.gstatic.com/amphtml/stamp/qa/sharing.html)
* [Tooltip](https://www.gstatic.com/amphtml/stamp/qa/tooltip.html)
* [Page attachment](https://www.gstatic.com/amphtml/stamp/qa/page-attachment.html)
* [Branching](https://www.gstatic.com/amphtml/stamp/qa/branching.html#page=page2)
* [Consent](https://www.gstatic.com/amphtml/stamp/qa/consent.html)
* [Prerender - Image](https://www.gstatic.com/amphtml/stamp/qa/prerender-img.html?amp_js_v=0.1#origin=https://www.google.com&prerenderSize=1&visibilityState=prerender&paddingTop=0&p2r=0&horizontalScrolling=0&csi=1&aoh=15526927198732&viewerUrl=https://www.google.com&history=1&storage=1&cid=1&cap=swipe,navigateTo,cid,fragment,replaceUrl)
* [Prerender - Video](https://www.gstatic.com/amphtml/stamp/qa/prerender-video.html?amp_js_v=0.1#origin=https://www.google.com&prerenderSize=1&visibilityState=prerender&paddingTop=0&p2r=0&horizontalScrolling=0&csi=1&aoh=15526927198732&viewerUrl=https://www.google.com&history=1&storage=1&cid=1&cap=swipe,navigateTo,cid,fragment,replaceUrl)
* [AMP Story Player - Player](https://www.gstatic.com/amphtml/stamp/qa/player.html)
* [AMP Story Player - AMP Story 360](https://www.gstatic.com/amphtml/stamp/qa/amp-story-360.html)

## Description
Browsers have functionality that lets people zoom in on the page, enlarging all of the content in equal proportion. If zoom hasn't been taken into consideration when designing the page, it can result in content overlapping or being obscured.

All tested samples display a "Expand the height/the width/both the height and width of your window to view this experience" message (rather than any of the actual page content) when the CSS dimensions of the viewport fall below particular thresholds (width above 768 CSS px and below 1024 CSS px, height below 550 CSS px, and possibly other combinations/ratios). As zooming in desktop browsers (to magnify/increase the size of the content) also changes the reported CSS width/height of the browser's viewport, this means that users that use zoom, as well as users that have a desktop environment running at a lower resolution, may end up seeing these "door slam" messages rather than the page content.

!['Expand the width of your window to view this experience' message shown in a browser with a width just below 1024px](https://user-images.githubusercontent.com/54933393/107247192-ee0d0980-6a28-11eb-8c51-5b2d52cb2522.png)
!['Expand the height of your window to view this experience' message shown in a browser with a height just below 550px](https://user-images.githubusercontent.com/54933393/107247246-fb29f880-6a28-11eb-8bb6-8964d475484d.png)

For instance, the pages are currently not usable for a user running a browser at 1920x1080 at 200% zoom (the requirement for [WCAG 2.1 Success Criterion 1.4.4 Resize Text (Level AA)](https://www.w3.org/TR/WCAG21/#resize-text)).

!['Expand both the height and width of your window to view this experience' message shown in browser at 1920x1080 at 200% zoom](https://user-images.githubusercontent.com/54933393/107247322-0e3cc880-6a29-11eb-8209-1eb89340d52d.png)

Further, this also leads to a failure of [WCAG 2.1 Success Criterion 1.4.10 Reflow (Level AA)](https://www.w3.org/TR/WCAG21/#reflow) - while content does "reflow" to 320 CSS px width, this only holds true if the height of the viewport is relatively high. For the actual use case that lies at the core of the success criterion - low vision desktop users that need the ability to view content even at high zoom factors such as 400% on a small 1280x1024 screen - all samples fail, since they show the "Expand both the height and width..." message instead of the page content.

!['Expand both the height and width of your window to view this experience' message shown in browser at 1280x1024 at 400% zoom](https://user-images.githubusercontent.com/54933393/107247377-1e54a800-6a29-11eb-9970-1fb2bb933551.png)

## User impact
When content is hidden when the page is viewed at zoom, it means someone with low vision who depends on zoom functionality to be able to read the content will not be able to do so.

## Required solution
Make sure that page content responds well to changes in zoom levels. In particular, make sure that page content does not overlap itself or get clipped when resized.

This solution must be applied to all instances of the issue identified within the test sample, then applied to all other instances of the same issue identified throughout the rest of the website.

## Implementation guidance

While it may be technically complex, AMP stories should ensure that they present users with content/functionality even in a small viewport.

## Test procedure(s)
Use these steps to confirm that the solution has been correctly applied to issues identified within the test sample, and to test the rest of the website for instances of the same issue:
1. Open the page in a browser.
2. Zoom content to at least 200%.
3. Check that all content and functionality is available without being obscured or overlapped.
4. Set the size of the browser to a "standard" desktop size (approximately 1280px wide).
5. Zoom content to at least 400%.
6. Check that the content reflows to a single column and that it can be viewed without both horizontal and vertical scrolling being necessary.

### Definition of done
Complete all of these tasks before closing this issue or indicating it is ready for retest:
* All issues identified within the test sample have been resolved.
* The rest of the website has been tested for the same issue.
* All issues identified throughout the rest of the website have been resolved or filed as new issues.

## Related standards
* [WCAG 2.1 Success Criterion 1.4.4 Resize Text (Level AA)](https://www.w3.org/TR/WCAG21/#resize-text)
* [WCAG 2.1 Success Criterion 1.4.10 Reflow (Level AA)](https://www.w3.org/TR/WCAG21/#reflow)

## More information
* [W3C: Using a technology that has commonly-available user agents that support zoom](https://www.w3.org/WAI/WCAG21/Techniques/general/G142.html)
* [W3C: Technique F102: Failure of Success Criterion 1.4.10 due to content disappearing and not being available when content has reflowed](https://www.w3.org/WAI/WCAG21/Techniques/failures/F102)
* [Knowbility: Exploring WCAG 2.1: 1.4.10 Reflow](https://knowbility.org/blog/2018/WCAG21-1410Reflow/)

## Test data
Test date: January 2021
Website:

**Comment by processprocess** [comment]
Fixed as part of #35267
